com.unity.ml-agents/Runtime/Demonstrations/DemonstrationRecorder.cs

```diff
@@ -70,6 +76,14 @@ void Update()
             {
                 LazyInitialize();
             }
+
+            if (m_DemoWriter.NumSteps > 0 && m_DemoWriter.NumSteps  >= NumStepsToRecord)
```

Comment:
```suggestion
            if (NumStepsToRecord > 0 && m_DemoWriter.NumSteps  >= NumStepsToRecord)
```
Not sure the first part will accomplish anything as it's written now.

Reply:
yeah caught that as well. it's fixed now.